packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+/// equivalent of Android's `ListView` and `RecyclerView`. [CupertinoTableView]
+/// is Flutter's iOS-styled alternative to [ListView].
+///
+/// #Cells
+///
+/// To achieve complete iOS style parity, consider filling [CupertinoTableView]s
+/// with [CupertinoTableViewCell]s. If custom cells are desired, any widget
+/// can be added to a [CupertinoTableView].
+///
+/// #Dividers
+///
+/// By default, [CupertinoTableView]s render dividers between each cell. For
+/// custom cells, the divider beneath the cell is rendered with a standard inset
+/// on the starting edge and is then drawn all the way to the ending edge. For
+/// [CupertinoTableViewCell]s, the divider starts where the title text starts
+/// and then renders all the way to the ending edge. This divider behavior is
+/// as per iOS behavior.
+///
+/// If a [CupertinoTableView] does not contain enough cells to fill the viewport
+/// then extra dividers are rendered beyond the last cell to the end of the
+/// viewport. This extra divider rendering extends into overscroll space at the
+/// end, too. The gap between extra dividers is equal to the height of the
+/// last cell in the [CupertinoTableView].
+///
+/// #Styles
+///
+/// A [CupertinoTableView] may be rendered either in a "plain" style or a
+/// "grouped" style.
+///
+/// A plain [CupertinoTableView] renders a single set of cells without any
+/// divisions between cells.
+///
+/// A grouped [CupertinoTableView] subdivides cells into sections. Sections
+/// are rendered with gaps between them and optional headers before each section.
+///
+/// #Section Index
+///
+/// If a [CupertinoTableView] is in a grouped style, it can optionally display
+/// a section "index". This index is a thin strip that appears on the right side
+/// of the [CupertinoTableView], displays titles for each section in the
+/// [CupertinoTableView], and allows users to tap on a given section title to
+/// automatically scroll to that section.
+///
+/// See also:
+///
+///  * [CupertinoTableViewCell], which renders an iOS-style table view cell that
+///    is intended to appear within a [CupertinoTableView].
+///  * [ListView], which renders a style agnostic list.
+class CupertinoTableView extends StatefulWidget {
+  /// Creates a plain-style [CupertinoTableView] with an explicit list of cells.
+  const CupertinoTableView.plain({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildren = const <Widget>[],
+  }) : plainChildrenBuilder = null,
+       plainChildCount = null;
+
+  /// Creates a plain-style [CupertinoTableView] with a lazily built list of cells.
+  const CupertinoTableView.plainBuilder({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildCount,
+    @required this.plainChildrenBuilder,
+  }) : plainChildren = null;
+
+  /// ScrollController that controls the scroll position of this [CupertinoTableView].
```

Comment:
Describe nullability and defaults

Reply:
I'm not sure what we should say here for that.  Any preference?  It can be null, in which case things work just like any CustomScrollView that either isn't given a ScrollController or is given a null ScrollController.

Reply:
The API doc for every field are their own page. Users can't see a larger pattern that exists in other widgets. 

Verbosity that are in other ScrollController field docs should be here.

Reply:
@xster what do you think about just using the macro doc inclusion from CustomScrollView?

Reply:
Having a macro for ScrollController around the codebase sounds good